packages/@aws-cdk/aws-codecommit/lib/pipeline-action.ts

```diff
@@ -54,11 +54,15 @@ export class PipelineSourceAction extends codepipeline.SourceAction {
       configuration: {
         RepositoryName: props.repository.repositoryName,
         BranchName: props.branch || 'master',
-        PollForSourceChanges: props.pollForSourceChanges !== undefined ? props.pollForSourceChanges : true
+        PollForSourceChanges: props.pollForSourceChanges || false,
       },
       outputArtifactName: props.outputArtifactName
     });
 
+    if (props.pollForSourceChanges !== true) {
+      props.repository.onCommit('CodePipeline', props.stage.pipelineAsEventTarget, props.branch || 'master');
```

Comment:
Why not just `stage.pipeline`?

Reply:
See above - it would cause a circular dependency. The entire reason for breaking out the `codepipeline-api` module is because Actions *cannot* reference the `codepipeline` module, where the `Pipeline` type lives.